odl/contrib/shearlab/shearlab_operator.py

```diff
@@ -0,0 +1,324 @@
+
+                    See Also
+                    --------
+                    odl.contrib.shearlab.ShearlabOperator
+                    """
+
+                    def __init__(self):
+                        """Initialize a new instance.
+                        Parameters
+                        ----------
+                        op : `ShearlabOperator`
+                              The operator which this should be the
+                              inverse of the adjoint of.
+                        """
+                        super(ShearlabOperatorAdjointInverse, self).__init__(
+                            op.domain, op.range, True)
+
+                    def _call(self, x):
+                        """``self(x)``."""
+                        with op.mutex:
+                            result = shearrecadjoint2D(x, op.shearlet_system)
+                            return np.moveaxis(result, -1, 0)
+
+                    @property
+                    def adjoint(self):
+                        """The adjoint operator."""
+                        return self.op.inverse
+
+                    @property
+                    def inverse(self):
+                        """The inverse operator."""
+                        return self.op.adjoint
+
+        return ShearlabOperatorAdjoint()
+
+    @property
+    def inverse(self):
+        """The inverse operator."""
+        op = self
+
+        class ShearlabOperatorInverse(odl.Operator):
+
+            """Inverse of the shearlet transform.
+
+            See Also
+            --------
+            odl.contrib.shearlab.ShearlabOperator
+            """
+
+            def __init__(self):
+                """Initialize a new instance.
+                Parameters
+                ----------
+                op : `ShearlabOperator`
+                    The operator which this should be the inverse of.
+                """
+                super(ShearlabOperatorInverse, self).__init__(
+                    op.range, op.domain, True)
+
+            def _call(self, x):
+                """``self(x)``."""
+                with op.mutex:
+                    x = np.moveaxis(x, 0, -1)
+                    return shearrec2D(x, op.shearlet_system)
+
+            @property
+            def adjoint(self):
+                """The adjoint operator."""
+                return ShearlabOperatorAdjointInverse()
+
+            @property
+            def inverse(self):
+                """The inverse operator."""
+                return op
+
+        return ShearlabOperatorInverse()
+
+
+# Python library for shearlab.jl
+
+def load_julia_with_Shearlab():
+    # Function to load Shearlab
+    # Importing base
+    j = julia.Julia()
+    j.eval('using Shearlab')
+    j.eval('using PyPlot')
+    j.eval('using Images')
+    return j
+
+
+j = load_julia_with_Shearlab()
+
+
+def load_image(name, n, m=None, gpu=None, square=None):
+    # Function to load images with certain size
```

Comment:
I wasn't clear enough. The comments should go into the function body *as docstrings*